src/Integration/Exclusions/AnalyzableFileIndicator.cs

```diff
@@ -87,6 +87,10 @@ public bool ShouldAnalyze(string filePath)
             exclusions != null &&
             exclusions.Any(x => IsMatch(x, filePath));
 
-        private bool IsMatch(string pattern, string filePath) => globPatternMatcher.IsMatch(pattern, filePath);
+        private bool IsMatch(string pattern, string filePath)
+        {
```

Comment:
IsMatch would be called multiple times, why do it each time? Not to mention it would be incorrect for the (newly added) debug logs.

I think it should be done at the start of the ShouldAnalyze method.